### ISSUE TITLE
Logind stuff in swayidle could be replaced with systemd-inhibit

### ISSUE THREAD

**Issue by emersion** [swaywm/sway#2215]
https://www.freedesktop.org/software/systemd/man/systemd-inhibit.html

**Comment by SirCmpwn** [comment]
Would that work with elogind?

**Comment by emersion** [comment]
Yes: https://github.com/elogind/elogind/blob/278eff0f0b60eda03a3775a28d2d4a3939eb4eba/meson.build#L1926

**Comment by SirCmpwn** [comment]
Coolio

**Comment by emersion** [comment]
Would probably need to have a `-d` flag to ask swaylock to fork in the background when it's ready. That would be cleaner than the current solution (which is to block inhibition for some time).

**Comment by emersion** [comment]
Hmm, after looking at this again, it seems that it won't be possible to remove dbus. While `systemd-inhibit` would work great to inhibit before going to sleep, we still need to listen to the logind sleep event to trigger the command.

Someone suggested listening to the lock event, considering all of this I'd say it's not out of scope for swayidle since it already listens to the sleep event...

**Comment by aereaux** [comment]
I have something written up for listening to the lock event, but it might need a bit more polishing.  I think either it should listen for both or neither, but I'm not sure which would be preferred.  I think @SirCmpwn had preferred neither (on IRC), but I'm not sure.

**Comment by emersion** [comment]
Yeah, right now it's inconsistent and the suspend event has nothing to do with the user being idle (even if being idle can trigger suspend). Also, it's possible to configure a systemd service to run swaylock on suspend (https://wiki.archlinux.org/index.php/Power_management#Sleep_hooks). Though, it's not possible to run swaylock on a logind lock event without a dedicated daemon.

As you said, we should choose one strategy and stick to it.

**Comment by emersion** [comment]
@SirCmpwn ping?

**Comment by SirCmpwn** [comment]
Yes?

**Comment by emersion** [comment]
Do you want to (1) remove dbus from swayidle or (2) add lock support? See comments above < 3 days old.

**Comment by SirCmpwn** [comment]
I would lean towards (1) but I don't really care so long as dbus stays optional. I don't use swayidle.
